<!DOCTYPE html>
<html>
<head>
  <title>UVM Proteomics Bioinformatics</title>
</head>
<body>
  <h1></h1>
  <ul>
    <li><a href="index.html">Home</a></li>
    <li><a href="MotifExtraction.html">Motif Extraction</a></li>
    <li><a href="page2.html">Page 2</a></li>
    <li><a href="page3.html">Page 3</a></li>
  </ul>
  
  <figure>
    <img src="/logo_landscape.png" alt="" />
    <figcaption aria-hidden="true">VBRN Logo</figcaption>
  </figure>
  
  <h1 id="uvm-proteomics"><a href="https://vbrn.org/proteomics/">UVM Proteomics</a></h1>
  
  <h3 id="authors">Authors</h3>
  <p><a href="https://scholar.google.com/citations?user=uGHWHbgAAAAJ&amp;hl=en&amp;oi=ao">Emily Curd</a> and James (Jax) Lubkowitz</p>
  
  <h3 id="affiliations">Affiliations</h3>
  <p>
    <a href="https://vbrn.org">Vermont Biomedical Research Network</a>,
    <a href="https://www.med.uvm.edu/vigr/home">Vermont Integrated Genomics Resource</a>,
    <a href="https://www.uvm.edu">University of Vermont</a>,
    <a href="https://www.stlawu.edu">St. Lawrence University</a>
  </p>
  
  <h3 id="funding">Funding</h3>
  <p>
    Research reported in this repository was supported by an Institutional Development Award (IDeA)
    from the National Institute of General Medical Sciences of the National Institutes of Health under
    grant number P20GM103449. Its contents are solely the responsibility of the authors and do not
    necessarily represent the official views of NIGMS or NIH.
  </p>
  
  <p>
    This is a R package for the <a href="https://vbrn.org/proteomics/">UVM - VBRN Proteomics Core</a>.
    It includes functions for three main types of analyses: Post Translational Modification, Spectral Counting,
    and Protein Expression.
  </p>
  
  <p><strong><em>Currently Supports or In Development</em></strong></p>
  
  <p>Post Translational Modification: - Motif Analysis</p>
  <p>Spectral Counting: - Differential protein expression (DEqMS)</p>
  <p>Protein Expression: - Pathway Analysis - functional classification</p>
  
  <h3 id="installation">Installation</h3>
  <p>
    Package can be installed from github via
  </p>
  
  <div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r">
install.packages("devtools")
devtools::install_github("UVM_Proteomics", username="jaxlub")
</code></pre></div>
  
  <p>
    Another download option is to download a zip file from github, unzip it and then run
  </p>
  
  <div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r">
install.packages("devtools")
source <- devtools:::source_pkg("path/to/package")
install(source)
</code></pre></div>
  
  <h1 id="post-translational-modification">Post Translational Modification</h1>
  
  <h2 id="motif-extraction">Motif Extraction</h2>
  
  <p>
    This function extracts predetermined protein motifs from a master protein in a user specified proteome
    (e.g., <a href="https://www.uniprot.org/proteomes?facets=proteome_type:1&amp;query=(organism_id:10090)">
    UniProt mouse proteome</a>). The function identifies the position of the modified amino acid(s) in a master protein,
    and uses that information as a midpoint to calculate the position range of amino acids in a motif. The blast+ suite
    is then used to search a proteome and extract the motifs. To do this, a user-defined proteome is converted to a blast
    searchable database (using makeblastdb). The master protein accession is used to search the database (using blastdbcmd),
    and if present, the motif determined by the range calculated above is extracted and appended to the xlsx table.
  </p>
  
  <p>
    If multiple master proteins or multiple modified amino acids are identified for a given row of input data,
    this function will separate that data to search a single master protein / modification at a time resulting in
    a greater number of rows in the data set.
  </p>
  
  
  
  <h3 id="the-output
